src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , initialSeparator] = fromFormat;
+  const [, , , resultingSeparator] = toFormat;
+  const initialArray = fromFormat.slice(0, 3);
+  const resultingArray = toFormat.slice(0, 3);
+  const dateArray = date.split(initialSeparator);
+  const result = [];
+
+  for (const initialElement of initialArray) {
+    for (const resultingElement of resultingArray) {
+      const initialIndex = initialArray.indexOf(initialElement);
```

Comment:
The `indexOf` method is being called inside a loop, which can lead to performance issues because it will search the entire array every time it is called. Since the loop is already iterating over `initialArray`, you already have the index as the loop iterator. You should use this index directly instead of calling `indexOf`.